ls1intum/Artemis
# Use JUnit 5 for the JVM Tests

### Checklist

### Motivation and Context
Switch to JUnit 5 and newer Maven Surefire version to support https://github.com/ls1intum/artemis-java-test-sandbox and also resolves #2273.

The name of the test cases in Artemis will **not** change.

### Description
- Use `artemis-java-test-sandbox` already as provider for JUnit 5, AssertJ and more. Remove JUnit 4.
- Use Maven Surefire version `3.0.0-M5`
- Adjust the Java test template to JUnit 5 Jupiter using dynamic tests.

#### TODO:
- [x] Change version and check some JavaDoc
- [x] Also update Kotlin (and #2278 is also affected)

### Steps for Testing
0. Review the code (note that the Kotlin code is the same apart from `FunctionalTest`)
1. Create a Java (and also Kotlin) programming exercise as instructor (with or without SCA should not matter) on TS1/TS3
2. Check that both solution and template have **13** test cases, with the solution passing all of them and the template none
3. Participate in some way and check that the tests still work. Check passing at least _one behavior test_ (e.g. sorting functions) and _one structural_ (e.g. adding the sort interface).
4. Check that the tasks **and** the class diagram in the problem statement are updated correctly
5. _Repeat that on TS3 which uses another build system (there might still be problems due to #2273 and #2224 regarding failing test)_

### Test Coverage

_Unchanged._
